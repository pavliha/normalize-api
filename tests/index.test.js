// eslint-env jest
import normalize from '../src'

describe('normalize', () => {
  test('normalizes entities  with nested objects', () => {

    const post = {
      id: 1,
      title: 'My first post',
      user: { id: 4, name: 'Post Writer' },
    }

    const entities = normalize(post, 'post')
    expect(entities).toMatchSnapshot()
  })

  test('normalizes entities arrays of objects', () => {

    const post = {
      id: 1,
      comments: [
        {
          id: 1,
          message: 'First comment',
          post_id: 1,
        },
        {
          id: 2,
          message: 'Second comment',
          post_id: 1,
        },
        {
          id: 3,
          message: 'Third comment',
          post_id: 1,
        }
      ]
    }

    const entities = normalize(post, 'post')
    expect(entities).toMatchSnapshot()
  })

  test('normalizes entity with arrays of objects inside and nested objects inside of array of objects', () => {

    const post = {
      id: 1,
      comments: [
        {
          id: 1,
          message: 'First comment',
          post_id: 1,
          user: { id: 1, name: 'User 1' }
        },
        {
          id: 2,
          message: 'Second comment',
          post_id: 1,
          user: { id: 2, name: 'User 2' }
        },
        {
          id: 3,
          message: 'Third comment',
          post_id: 1,
          user: { id: 2, name: 'User 2' }
        }
      ]
    }

    const entities = normalize(post, 'post')
    expect(entities).toMatchSnapshot()
  })

  test('normalizes entities arrays of objects', () => {

    const posts = [
      {
        id: 1,
        message: 'First comment',
        post_id: 1,
        user: { id: 1, name: 'User 1' }
      },
      {
        id: 2,
        message: 'Second comment',
        post_id: 1,
        user: { id: 2, name: 'User 2' }
      },
      {
        id: 3,
        message: 'Third comment',
        post_id: 1,
        user: { id: 2, name: 'User 2' }
      }
    ]

    const entities = normalize(posts, 'post')
    expect(entities).toMatchSnapshot()
  })
})


