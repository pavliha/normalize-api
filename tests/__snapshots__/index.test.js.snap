// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize normalizes entities  with nested objects 1`] = `
Object {
  "post": Array [
    Object {
      "id": 1,
      "title": "My first post",
      "user_id": 4,
    },
  ],
  "user": Array [
    Object {
      "id": 4,
      "name": "Post Writer",
    },
  ],
}
`;

exports[`normalize normalizes entities arrays of objects 1`] = `
Object {
  "comments": Array [
    Object {
      "id": 1,
      "message": "First comment",
      "post_id": 1,
    },
    Object {
      "id": 2,
      "message": "Second comment",
      "post_id": 1,
    },
    Object {
      "id": 3,
      "message": "Third comment",
      "post_id": 1,
    },
  ],
  "post": Array [
    Object {
      "id": 1,
    },
  ],
}
`;

exports[`normalize normalizes entities arrays of objects 2`] = `
Object {
  "post": Array [
    Object {
      "id": 3,
      "message": "Third comment",
      "post_id": 1,
      "user_id": 2,
    },
    Object {
      "id": 2,
      "message": "Second comment",
      "post_id": 1,
      "user_id": 2,
    },
    Object {
      "id": 1,
      "message": "First comment",
      "post_id": 1,
      "user_id": 1,
    },
  ],
  "user": Array [
    Object {
      "id": 2,
      "name": "User 2",
    },
    Object {
      "id": 2,
      "name": "User 2",
    },
    Object {
      "id": 1,
      "name": "User 1",
    },
  ],
}
`;

exports[`normalize normalizes entity with arrays of objects inside and nested objects inside of array of objects 1`] = `
Object {
  "comments": Array [
    Object {
      "id": 1,
      "message": "First comment",
      "post_id": 1,
      "user_id": 1,
    },
    Object {
      "id": 2,
      "message": "Second comment",
      "post_id": 1,
      "user_id": 2,
    },
    Object {
      "id": 3,
      "message": "Third comment",
      "post_id": 1,
      "user_id": 2,
    },
  ],
  "post": Array [
    Object {
      "id": 1,
    },
  ],
  "user": Array [
    Object {
      "id": 1,
      "name": "User 1",
    },
    Object {
      "id": 2,
      "name": "User 2",
    },
  ],
}
`;
